pennylane_lightning/lightning_qubit.py

```diff
@@ -617,10 +608,15 @@ def expval(self, observable, shot_range=None, bin_size=None):
         Returns:
             Expectation value of the observable
         """
-        if isinstance(observable.name, List) or observable.name in [
-            "Identity",
-            "Projector",
-        ]:
+        if (
+            (observable.arithmetic_depth > 0)
```

Comment:
What is the `arithmetic_depth` of an observable?

Reply:
Arithmetic depth is the level of nesting. So `Sum(qml.PauliX(0), qml.PauliY(0))` will have an arithmetic depth of 1.  `Sum(adjoint(qml.PauliX(0)), qml.PauliY(0) ))` will have an arithmetic depth of 2.  Plain elementary ops will have an arithmetic depth of `0`.

Reply:
I assume then this is pulled as the height of the tree representing the operation?

While not an issue here, just clarifying so that we can provide a way to port this to CPP later.

Reply:
Yes, it is defined for all `Operator` objects.  And yes, it is just the height of the tree.